### DISCUSSION TITLE
auth-via-email late response

### DISCUSSION THREAD

**Discussion by waterdrw** [pocketbase/pocketbase#298] [Q&A]
hi! i really love your works 
anyway i'm using pocketbase our small project.
auth-via-email really late response both pocketbase dashboard and my frontend 
![image](https://user-images.githubusercontent.com/912132/184292515-f15e3739-67fa-4859-9468-4a5728c0796d.png)
and no error or warning messages in logs 

everything else is fine 👍 
what should i check?

thank you!

**Comment by ganigeorgiev** [comment]
Hm, the only thing that I can think of could result in slow server response is the email validation, since we are checking also the MX record of the domain.

You could try using `dig`, `nslookup`, `host`, etc. commands to check how long it takes to retrieve the dns information for your email domain. For example:
```sh
host -t MX gmail.com
```

**Comment by ganigeorgiev** [comment]
I wasn't able to reproduce the slow response time. In the past another user also reported something similar with slow start up times on Windows - https://github.com/pocketbase/pocketbase/issues/6#issuecomment-1178895561, so maybe it is related.

_I'll move the issue to Q&A discussions since the format is more suitable for that._

**Comment by waterdrw** [comment]
i'm serve in ubuntu 16.04 :) 

MX record result here
![image](https://user-images.githubusercontent.com/912132/184306442-3f3de91a-6176-43fd-a05d-996a7a3701c3.png)

i serve --http="127.0.0.1" --debug mode. but no errors or warnings

**Reply by ganigeorgiev** [reply]
1. Is the PocketBase slow server response always reproducable when trying to login?
2. Do you have any firewall or other proxy that could interfere with the request-response lifecycle?
3. Can you reproduce the issue locally or only when deploying on prod environment?
4. Could you try creating a dummy `test@example.com` admin and try to login with it?

**Reply by waterdrw** [reply]
1. yes. login and add new member. i think too this problem cause by email validation
2. my server under VPN but dont block anything. just using it tunnering other server  
3. my local PB no problem. 
4. yeah it can create and fast response login 

sorry for delayed response ;)

**Reply by ganigeorgiev** [reply]
Since there is no delay when using `test@example.com`, I'm mostly sure that this is caused by the VPN as a result of a slow MX email domain lookup (`example.com` is excluded from the check since it is IANA-managed and it is commonly used for development).

To confirm it, you could try running the same `host/nslookup` command against your email domain using the VPN and check if the result is returned with some delay.

We can remove the additional email domain validation on login, but I see very little benefit to that because you'll need it anyway during registration. From personal experience, checking only the email format is not enough since sending to missing or wrong emails (eg. with typos in the domain) will increase your bounce rate and could hurt your mail server reputation and overall deliverability.

**Reply by ganigeorgiev** [reply]
@waterdrw I've disabled the MX domain lookup **only for user and admin login** in the latest [v0.5.0 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.5.0). The standard email validation is still active when you create a new user/admin account.
